pkg/apis/servicecatalog/types.go

```diff
@@ -56,15 +56,40 @@ type BrokerSpec struct {
 // BrokerAuthInfo is a union type that contains information on one of the authentication methods
 // the the service catalog and brokers may support, according to the OpenServiceBroker API
 // specification (https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md).
-//
-// Note that we currently restrict a single broker to have only one of these fields
-// set on it.
 type BrokerAuthInfo struct {
-	// BasicAuthSecret is a reference to a Secret containing auth information the
-	// catalog should use to authenticate to this Broker using basic auth.
-	BasicAuthSecret *v1.ObjectReference
+	// SecretRef is a reference to a Secret containing auth information the
+	// catalog should use to authenticate to this Broker using corresponding auth.
+	SecretRef *v1.ObjectReference
+	// Used to facilitate programmatic handling of auth secret data.
+	// Defaults to "basic"
+	// +optional
+	SecretType BrokerAuthSecretType
 }
 
+// BrokerAuthSecretType is an identifier of Secret type to be used for authentication
+type BrokerAuthSecretType string
+
+const (
+	// SecretTypeBasicAuth contains data needed for basic authentication.
+	//
+	// Required at least one of fields:
+	// - Secret.Data["username"] - username used for authentication
+	// - Secret.Data["password"] - password or token needed for authentication
+	SecretTypeBasicAuth BrokerAuthSecretType = "basic"
+	// BasicAuthUsernameKey is the key of the username for SecretTypeBasicAuth secrets
+	BasicAuthUsernameKey = "username"
+	// BasicAuthPasswordKey is the key of the password or token for SecretTypeBasicAuth secrets
+	BasicAuthPasswordKey = "password"
+
+	// SecretTypeBearer contains data needed for bearer token authentication.
+	//
+	// Required field:
+	// - Secret.Data["token"] - bearer token for authentication
+	SecretTypeBearer BrokerAuthSecretType = "bearer"
```

Comment:
Naming suggestion: `SecretTypeBearerTokenAuth`, so that it's more descriptive and is consistent with the `Auth` suffix above. I won't push any further than this on naming; feel free to ignore if you don't like it 😁